amundsen-kube-helm/templates/helm/templates/amundsen-oidc-config.yaml

```diff
@@ -11,14 +11,33 @@ stringData:
       "web": {
         "client_id": "{{ .Values.frontEnd.OIDC_CLIENT_ID }}",
         "client_secret": "{{ .Values.frontEnd.OIDC_CLIENT_SECRET }}",
+
+        {{- if eq .Values.frontEnd.OIDC_PROVIDER "azure" }}
+        "auth_uri": "{{ .Values.frontEnd.OIDC_ORG_URL }}/oauth2/v2.0/authorize",
+        "token_uri": "{{ .Values.frontEnd.OIDC_ORG_URL }}/oauth2/v2.0/token",
+        "issuer": "{{ .Values.frontEnd.OIDC_ORG_URL }}/v2.0",
+        "userinfo_uri": "https://graph.microsoft.com/oidc/userinfo",
+
+        {{- else if eq .Values.frontEnd.OIDC_PROVIDER "google" }}
+        "auth_uri": "https://accounts.google.com/o/oauth2/v2/auth",
+        "token_uri": "https://oauth2.googleapis.com/token",
+        "issuer": "https://accounts.google.com",
+        "userinfo_uri": "https://openidconnect.googleapis.com/v1/userinfo",
```

Comment:
this looks good. long term, it might be better to move the logic around which uri's to use into an `oidc-config.py` or whatever is responsible for creating the secrets file. this is because google uri's will change , which would then require updating the helm chart every time an endpoint changes. you can always get the latest uri's from google from `https://accounts.google.com/.well-known/openid-configuration`. i might take a stab at this in a later PR

Reply:
Got it. Just wanted to go with the simpler static approach to get off the ground. 

I recently learned about Helm hooks - maybe they're useful to run the code`(would that be the right time/place). I think it's called OpenID Connect Discovery - there's already a tool for it here: 

https://github.com/puiterwijk/flask-oidc/blob/7f16e27b926fc12953d6b2ae78a9b9cc9b8d1769/flask_oidc/registration_util.py#L65

Oh, yeah and if you want to get cracking away on something you should actually be able to get this to login a user with Google, as I did, right now (there is just no validation of the token). Having that up and running on your end would mean you're prepared for testing the validation, figuring out other parts needed like documentation etc. and you have a demo environment.

Reply:
Great! thanks! I'll start playing around with it. I hacked around on a chart earlier and got it working with the frontend, but the problem I ran into was calling the metadata service and search service because of the lack of token validation. Once I created a dev Okta account it was working tho, so I think thats the only missing piece. I'll looking into the helm hooks too, I'd never heard of that

Reply:
Excellent! In the meantime I've got myself setup with a clone of needed intermediate pieces (Verdan's `flaskoidc` wrapper + `ammundsenfrontendlibrary` + local Kubernetes on Docker Desktop + Helm chart PR above) and a way to locally build ... 

* I'll try to get WiP PR's for that pushed for you (& possibly others) to see. Maybe also description of the build steps (it's a bit of Russian doll game due to all the layers, rebuild, testing, rinse & repeat 🙂)
* So far I've successfully swapped in `flask-oidc-ex` and added env vars from Damian's `dcrdev/flaskoidc` fork and my Google testing login still works with the frontend (without validation disabled as per Helm above)
* On the sad side even though I added Damian's CLOCK_SKEW I'm still not able to login with Azure.
* Next up is adding the `offline` validation as a config option, then dig more into Azure login.

I guess with your Okta account you can cover testing that with the changes we introduce nothing gets broken with that provider. We need to get similar for Keycloak as well. 

I think we should also move out discussion from here and create a proper issue to track across PRs and get more visibility.